python/lsst/ts/Dome/dome_csc.py

```diff
@@ -393,6 +400,22 @@ async def status(self):
         cmd = {"status": {}}
         await self.write(cmd)
         self.lower_level_status = await self.read()
+        self.tel_domeADB_status.set_put(
+            positionError=self.lower_level_status["AMCS"]["positionError"],
+            positionActual=self.lower_level_status["AMCS"]["positionActual"],
+            positionCmd=self.lower_level_status["AMCS"]["positionCmd"],
+            driveTorqueActual=self.lower_level_status["AMCS"]["driveTorqueActual"],
+            driveTorqueError=self.lower_level_status["AMCS"]["driveTorqueError"],
+            driveTorqueCmd=self.lower_level_status["AMCS"]["driveTorqueCmd"],
+            driveCurrentActual=self.lower_level_status["AMCS"]["driveCurrentActual"],
+            driveTempActual=self.lower_level_status["AMCS"]["driveTempActual"],
+            encoderHeadRaw=self.lower_level_status["AMCS"]["encoderHeadRaw"],
+            encoderHeadCalibrated=self.lower_level_status["AMCS"][
+                "encoderHeadCalibrated"
+            ],
+            resolverRaw=self.lower_level_status["AMCS"]["resolverRaw"],
+            resolverCalibrated=self.lower_level_status["AMCS"]["resolverCalibrated"],
+        )
```

Comment:
This works, but I worry about all the repetition. A spelling or copy/paste error could be hard to spot.

It appears that `self.lower_level_status["AMCS"]` is a dict. If *all* the values in that dict are fields in the telemetry event, then this would work:
```
self.tel_domeADB_status.set_put(**self.lower_level_status["AMCS"])
```
If you are only setting a subset of the fields from `self.lower_level_status["AMCS"]` then it takes more work.
You could list the fields manually, but if the public telemetry topic fields are a subset of the fields in `self.lower_level_status["AMCS"]` then you can make the list automatically:
```
# Put this in the constructor, so you only have to do it once
# (get_vars creates a dict representation of a DDS sample)
data = self.tel_domeADB_status_fields.DataType().get_vars()
self.domeADB_status_fields = {field for field in data if not field.startswith("private_")
# If Done is indexable then delete the field "DomeID" here

# Then in this code block:
adb_status_dict = {field: self.lower_level_status["AMCS"][field] for field in self.domeADB_status_fields}
self.tel_domeADB_status.set_put(**adb_status_dict)
```
The code isn't as clear, but it eliminates a lot of repetition.

Reply:
Thanks! I was worried about this as well. Fortunately, only one key in the dict needs to be taken out and the rest can stay. So I took a slightly different approach to simply copy the dict with one key less. I create a copy because I need to keep the original dict for internal purposes. And the code is written such that it is easy to expand the set of keys with more keys that need to be removed if necessary.

Reply:
The new code looks great. Nice solution.